casanovo/data/ms_io.py

```diff
@@ -2,18 +2,56 @@
 
 import collections
 import csv
+import dataclasses
 import operator
 import os
 import re
 from pathlib import Path
-from typing import List
+from typing import List, Tuple
 
 import natsort
 
 from .. import __version__
 from ..config import Config
 
 
+@dataclasses.dataclass
+class PepSpecMatch:
+    """
+    Peptide Spectrum Match (PSM) dataclass
+
+    Parameters
+    ----------
+    sequence : str
+        The amino acid sequence of the peptide.
+    spectrum_id : Tuple[str, str]
+        A tuple containing the spectrum identifier in the form
+        (spectrum file name, spectrum file idx)
+    peptide_score : float
+        Confidence score of the match between the full peptide sequence and the
+        spectrum.
+    charge : int
+        The charge state of the peptide ion observed in the spectrum.
+    expected_mz : float
```

Comment:
**todo:** I suggest renaming this one and the next to `calc_mz` (calculated) and `exp_mz` (experimental) in analogy to the mzTab output columns.